### ISSUE TITLE
Invalid argument supplied for foreach()

### ISSUE THREAD

**Issue by yuchen99** [laravel/framework#7271]
A php script run background processes, first time Works perfect,but a day ago  throw,
Invalid argument supplied for foreach() .
 
code like 
```php
   $driver = Driver::remember(10,'driver_cache'.$driver_id)->find($driver_id);
```
Throws: Warning: Invalid argument supplied for foreach() in 
```
#0 /data//vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(455): Illuminate\Exception\Handler->handleError(2, 'Invalid argumen...', ...', 455, Array)
#1 /data/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(137): Illuminate\Database\Eloquent\Builder->getModels(Array)
#2 /data//vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(111): Illuminate\Database\Eloquent\Builder->get(Array)
#3 /data//vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(68): Illuminate\Database\Eloquent\Builder->first(Array)
y)
```

**Comment by dwightwatson** [comment]
Is this Laravel 4.2?

**Comment by yuchen99** [comment]
composer.json

 "require": {
        "laravel/framework": "4.1.*",
        "pda/pheanstalk": "dev-master",
        "nategood/httpful": "*"
    },

**Comment by GrahamCampbell** [comment]
Only laravel 4.2/5.0 is supported. Also `"pda/pheanstalk": "dev-master"` does not work on laravel 4.1.

**Comment by ammont** [comment]
@yuchen99 What is your cache driver? because we are experiencing the same issue.

**Comment by yuchen99** [comment]
@ammont memcached

**Comment by ammont** [comment]
@yuchen99 have you found a solution? We use redis.

**Comment by rvwoens** [comment]
Is fixed in 4.2; Problem was that tagged cache remember was non-atomic, which on a busy server leads to problems. In file Taggedcache.php:

    if ($this->has($key)) return $this->get($key);  // incorrect, non-atomic
    if ( ! is_null($value = $this->get($key))) return $value;  // correct, atomic

**Comment by munotsagar** [comment]
Hi 
Facing below issue, Could you please help me to sort out :

![composer_update](https://cloud.githubusercontent.com/assets/12947915/17973704/176ced58-6b01-11e6-9be1-3f6d75e4a429.JPG)
